smithy-rules-engine/src/main/java/software/amazon/smithy/rulesengine/language/util/SourceLocationHelpers.java

```diff
@@ -0,0 +1,84 @@
+/*
+ * Copyright 2022 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+package software.amazon.smithy.rulesengine.language.util;
+
+import software.amazon.smithy.model.SourceLocation;
+import software.amazon.smithy.utils.SmithyInternalApi;
+
+@SmithyInternalApi
+public final class SourceLocationHelpers {
```

Comment:
Not sure what the standard smithy practice is, not sure if things marked internal are also namespaced as such or not.